packages/components/src/components/data-table/data-table.hbs

```diff
@@ -223,8 +242,14 @@
               <td class="{{@root.prefix}}--data-table-cell--editable"><span class="{{@root.prefix}}--data-table-cell__content">{{data}}</span></td>
             <!-- all other td's -->
             {{else}}
-              <td {{#if column.numerical}} class="{{@root.prefix}}--data-table--numerical-cell" {{/if}}>
+              <td class="{{#if column.numerical}}{{@root.prefix}}--data-table--numerical-cell{{/if}}{{#if @root.isSmall}} td-small{{/if}}">
                 <!-- truncated new markup -->
+                {{#if @root.isSmall}}
+                  <label class="bx--label" for="{{column.title}}-{{@../index}}">
+                    {{column.title}}
+                  </label>
+                  <div id="{{column.title}}-{{@../index}}">
```

Comment:
`id` based on column title tend to cause ID duplication. A unique ID, probably auto-generated, should be used (e.g. as a prefix)

Reply:
fixed in https://github.com/carbon-design-system/carbon/pull/2992/commits/5fe1847a7841cd3c474cb413a8ed1b69a7cc618a